src/app/@core/directives/normal-character-only.directive.ts

```diff
@@ -0,0 +1,24 @@
+import { Directive, ElementRef, HostListener } from '@angular/core';
+
+@Directive({
+  selector: '[normalCharacterOnly]',
+})
+export class NormalCharacterOnlyDirective {
+  private regex: RegExp = new RegExp(/^[0-9a-zA-Z_ÀÁÂÃÈÉÊÌÍÒÓÔÕÙÚĂĐĨŨƠàáâãèéêìíòóôõùúăđĩũơƯĂẠẢẤẦẨẪẬẮẰẲẴẶẸẺẼỀỀỂưăạảấầẩẫậắằẳẵặẹẻẽềềểỄỆỈỊỌỎỐỒỔỖỘỚỜỞỠỢỤỦỨỪễệỉịọỏốồổỗộớờởỡợụủứừỬỮỰỲỴÝỶỸửữựỳỵỷỹ ]+$/);
```

Comment:
use this regex RegExp(/^[0-9a-zA-Z_\u00C0-\u1EF9 ]+$/);

Reply:
done